SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -386,6 +399,51 @@ TArray<Worker_InterestOverride> USpatialSender::CreateComponentInterest(AActor*
 	return ComponentInterest;
 }
 
+RPCsOnEntityCreation USpatialSender::PackQueuedRPCsForActor(TArray<TSharedRef<FPendingRPCParams>>* RPCList, AActor* Actor)
+{
+	RPCsOnEntityCreation QueuedRPCs;
+	check((RPCList != nullptr) && (Actor != nullptr));
+	if ((RPCList == nullptr) || (Actor == nullptr))
+	{
+		return QueuedRPCs;
+	}
+
+	UClass* Class = Actor->GetClass();
+	const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Class);
+
+	for (TSharedRef<FPendingRPCParams> RPCParams : *RPCList)
+	{
+		if (!RPCParams->TargetObject.IsValid())
+		{
+			// The target object was destroyed before we could send the RPC.
+			continue;
+		}
+
+		const FRPCInfo* RPCInfo = Info.RPCInfoMap.Find(RPCParams->Function);
+		FUnrealObjectRef TargetObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(PackageMap->GetNetGUIDFromObject(Actor)));
+		if (TargetObjectRef != FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
+		{
+			TSet<TWeakObjectPtr<const UObject>> UnresolvedObjects;
```

Comment:
Do you think there's maybe a way to group lines 426 to 437 into a separate function can make both this function and the function that sends RPCs normally, use it?

Reply:
Created a PackRPCDataToSpatialNetBitWriter function to remove code duplication.